src/libmongoc/src/mongoc/mongoc-host-list.c

```diff
@@ -303,7 +303,10 @@ _mongoc_host_list_from_hostport_with_err (mongoc_host_list_t *link_,
                                           bson_error_t *error)
 {
```

Comment:
Recommend adding `BSON_ASSERT_PARAM` for `host` and `link_` to guard subsequent dereferences.